src/client/js/partials/breaches.js

```diff
@@ -17,74 +17,91 @@ const state = new Proxy({
   }
 })
 
-let breachRows, emailSelect, statusFilter, resolvedCountOutput, unresolvedCountOutput
+let breachesTable, breachRows, emailSelect, statusFilter, resolvedCountOutput, unresolvedCountOutput
 
 function init () {
-  breachRows = breachesPartial.querySelectorAll('.breach-row')
+  breachesTable = breachesPartial.querySelector('.breaches-table')
+  breachRows = breachesTable.querySelectorAll('.breach-row')
   emailSelect = breachesPartial.querySelector('.breaches-header custom-select')
   statusFilter = breachesPartial.querySelector('.breaches-filter')
   resolvedCountOutput = statusFilter.querySelector("label[for='breaches-resolved'] output")
   unresolvedCountOutput = statusFilter.querySelector("label[for='breaches-unresolved'] output")
 
-  state.selectedEmail = emailSelect.value
+  state.selectedEmail = emailSelect.value // triggers render
+
   emailSelect.addEventListener('change', handleEvent)
   statusFilter.addEventListener('change', handleEvent)
-  render()
+  breachesTable.addEventListener('change', handleEvent)
 }
 
 function handleEvent (e) {
-  switch (e.currentTarget) {
-    case emailSelect:
+  switch (true) {
+    case e.target.matches('custom-select[name="email-account"]'):
       state.selectedEmail = e.target.value
       break
-    case statusFilter:
+    case e.target.matches('input[name="breaches-status"]'):
       state.selectedStatus = e.target.value
       break
+    case e.target.matches('.resolve-list-item [type="checkbox"]'):
+      updateBreachStatus(e.target)
+      break
```

Comment:
Suggestion: We should add a `default` case and throw an error if we fall through.

Reply:
Interesting suggestion.  I'm leaning towards leaving as is, because I can imagine a future case where we might add a listener to a parent element which might fire for elements we don't care about.  In this theoretical case we would not want to error (or do anything) – we'd want it to just pass through.

Reply:
The idea of my suggestion was that if we introduce another listener we can explicitly handle it which would make it easy to understand which events we care or do not care about. I’m totally fine with leaving it as is.